fvtest/jitbuildertest/JBTestUtil.hpp

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#ifndef JB_TEST_UTIL_HPP
+#define JB_TEST_UTIL_HPP
+
+#include "gtest/gtest.h"
+
+#include "Jit.hpp"
+#include "ilgen/TypeDictionary.hpp"
+#include "ilgen/MethodBuilder.hpp"
+
+/*
+ * Convenience macro for defining type dictionary objects. `name` is the name of
+ * the class that will be created. The "body" of the macro should contain the
+ * JitBuilder type definitions.
+ *
+ * Example use:
+ *
+ *    DEFINE_TYPES(MyTypeDictionary)
+ *       {
+ *       DefineStruct("MyStruct");
+ *       DefineField("MyStruct", "field", Int32);
+ *       CloseStruct("MyStruct");
+ *       }
+ */
+#define DEFINE_TYPES(name) \
+   struct name : public TR::TypeDictionary { name(); }; \
+   inline name::name() : TR::TypeDictionary()
+
+/*
+ * Convenience macro for declaring a MethodBuilder class. `name` is the name of
+ * the class that will be created. The macro will ensure the class inherits from
+ * `TR::MethodBuilder` and that the constructor and `buildIL()` method are
+ * declared. A ';' is required at the end of the macro invocation.
+ *
+ * Example use:
+ *
+ *    DECL_TEST_BUILDER(MyFunctionBuilder);
+ */
+#define DECL_TEST_BUILDER(name) \
+   class name : public TR::MethodBuilder { \
+      public: \
+      name(TR::TypeDictionary *); \
+      virtual bool buildIL(); \
+   }
+
+#define DEF_TEST_BUILDER_CTOR(name) \
+   inline name::name(TR::TypeDictionary *types) : TR::MethodBuilder(types)
```

Comment:
It's inline to avoid creating duplicate symbols in the (unlikely) event that the constructor is defined in a header file that gets included in multiple translation units.